### PR TITLE
Add cats.data.Ior

### PR DESCRIPTION
**Pull request by mpilquist**
Adds data type described in #273. Note: should be merged after #274.

### CODE REVIEW

File: data/src/main/scala/cats/data/Ior.scala [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+  )
+}
+
+object Ior extends IorInstances with IorFunctions {
+  final case class Left[+A](a: A) extends (A Ior Nothing)
+  final case class Right[+B](b: B) extends (Nothing Ior B)
+  final case class Both[+A, +B](a: A, b: B) extends (A Ior B)
+}
+
+sealed abstract class IorInstances {
```
Comment (xuwei-k) [inline data/src/main/scala/cats/data/Ior.scala]:
```scala
sealed abstract class IorInstances extends IorInstances0
```

**Comment by mpilquist** [comment]
@xuwei-k Thanks, good catch, fixed

**Comment by mpilquist** [comment]
Once I get this passing on Travis, I'll rebase to a single commit.

**Comment by non** [comment]
Looks good, thanks! :+1:

**Comment by stew** [comment]
yeah, looks good, thanks! :+1:

**Pull request merged**